inspectors/models/notation/fretdiagrams/fretdiagramsettingsmodel.h

```diff
@@ -0,0 +1,59 @@
+#ifndef FRETDIAGRAMSETTINGSMODEL_H
+#define FRETDIAGRAMSETTINGSMODEL_H
+
+#include "models/abstractinspectormodel.h"
+
+#include "libmscore/fret.h"
+#include "libmscore/score.h"
+
+class FretDiagramSettingsModel : public AbstractInspectorModel {
+    Q_OBJECT
+
+    Q_PROPERTY(PropertyItem* strings READ strings CONSTANT)
+    Q_PROPERTY(PropertyItem* frets   READ frets   CONSTANT)
+    Q_PROPERTY(PropertyItem* showNut READ showNut CONSTANT)
+    Q_PROPERTY(PropertyItem* offset  READ offset  CONSTANT)
+    Q_PROPERTY(PropertyItem* numPos  READ numPos  CONSTANT)
+
+public:
+    explicit FretDiagramSettingsModel(QObject* parent, IElementRepositoryService* repository);
+
+    void createProperties() override;
+    void requestElements() override;
+    void loadProperties() override;
+    void resetProperties() override;
+
+    PropertyItem* strings() const;
+    PropertyItem* frets() const;
+    PropertyItem* showNut() const;
+    PropertyItem* offset() const;
+    PropertyItem* numPos() const;
+
+    Q_INVOKABLE bool showFretCanvas() const;
+
+    Ms::FretDiagram* fretDiagram() const;
+    Ms::FretDiagram* fretDiagram();
+    Q_INVOKABLE int dot(int s, int f) const;
```

Comment:
in order to avoid of using so many Q_INVOKABLE's i'd suggest you to remember the scheme that i've sent to you

Let's decompose the complexity and create the "String" items. So the FretDiagramModel will manage the strings, that's all. Each string would manage its own properties like "barre", "dot", etc.